### PR TITLE
Fixed handling of long integers with qt mappings

### PR DESCRIPTION
**Pull request by fbridault**
This PR tries to fix the issue reported there https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=876147

The main issue was a a confusion between int and long types. Since `QVariant` does not have a long type, the long (64 bits on Unixes) was shortened to an int (32 bits on Unixes). The extra garbage was not a problem on little endian archs but was of course problematic on big endian archs.

So here is an attempt to fix this issue. On the Qt side, the proposed solution tries to use `QVariant::{U}LongLong` instead of `QVariant:{U}Int` when converting, depending on the size of long.

### CODE REVIEW

File: include/camp/valuemapper.hpp [hunk 1]
```diff
@@ -312,11 +324,13 @@ struct ValueMapper<camp::EnumObject>
...
-    static camp::UserObject from(double)                  {CAMP_ERROR(camp::BadType(camp::realType,   camp::enumType));}
-    static camp::UserObject from(const std::string&)      {CAMP_ERROR(camp::BadType(camp::stringType, camp::enumType));}
-    static camp::UserObject from(const camp::UserObject&) {CAMP_ERROR(camp::BadType(camp::enumType,   camp::enumType));}
+    static camp::UserObject from(bool)                    {CAMP_ERROR(camp::BadType(camp::boolType,     camp::enumType));}
+    static camp::UserObject from(int)                     {CAMP_ERROR(camp::BadType(camp::intType,      camp::enumType));}
+    static camp::UserObject from(long)                    {CAMP_ERROR(camp::BadType(camp::intType,      camp::enumType));}
+    static camp::UserObject from(long long)               {CAMP_ERROR(camp::BadType(camp::intType,      camp::enumType));}
+    static camp::UserObject from(double)                  {CAMP_ERROR(camp::BadType(camp::realType,     camp::enumType));}
+    static camp::UserObject from(const std::string&)      {CAMP_ERROR(camp::BadType(camp::stringType,   camp::enumType));}
```
Comment (luisjomen2a) [inline include/camp/valuemapper.hpp:332]: Too many spaces after the `,` character?

File: test/qt/propertymapping.cpp [hunk 2]
```diff
@@ -32,6 +32,8 @@
 #include <camp/classget.hpp>
 #include <boost/test/unit_test.hpp>
 
+#include <numeric>
+
```
Comment (luisjomen2a) [inline test/qt/propertymapping.cpp]: Too many empty lines?

File: include/camp/valuemapper.hpp (hunk 1 above)
Comment (fbridault) [reply include/camp/valuemapper.hpp:332]: That's just to keep everything aligned. And keep in mind we do not follow FW4SPL coding guidelines here, we just try to stick to the existing code.

File: test/qt/propertymapping.cpp (hunk 2 above)
Comment (fbridault) [reply test/qt/propertymapping.cpp]: Same remark.

**Review by schweitzer — APPROVED** [review]

**Review by luisjomen2a — APPROVED** [review]
Compiles on Windows 10

**Pull request merged**